packages/compile-vyper/index.js

```diff
@@ -178,53 +171,101 @@ async function compileAllNoJson({ sources, options, version }) {
+    }
   },
 
-  //since we don't have an imports analyzer for Vyper
-  //yet, we'll just treat this the same as all; this will
-  //need to be revisited once we have an import parser for Vyper
-  async sourcesWithDependencies({ paths: _paths = [], options }) {
-    return await Compile.all({ options });
+  async sourcesWithDependencies({ paths = [], options }) {
+    options = Config.default().merge(options);
+    debug("paths: %O", paths);
+    const vyperFilesStrict = paths.filter(path =>
+      minimatch(path, VYPER_PATTERN_STRICT, { dot: true })
+    );
+    debug("vyperFilesStrict: %O", vyperFilesStrict);
+
+    // no vyper targets found, no need to check Vyper
+    if (vyperFilesStrict.length === 0) {
+      return { compilations: [] };
+    }
+
+    const { allSources, compilationTargets } = await requiredSources(
+      options.with({
+        paths: vyperFilesStrict,
+        base_path: options.contracts_directory
+      })
+    );
+
+    debug("allSources: %O", allSources);
+    debug("compilationTargets: %O", compilationTargets);
+    const vyperTargets = compilationTargets.filter(path =>
+      minimatch(path, VYPER_PATTERN_STRICT, { dot: true })
+    );
+
+    // no vyper targets found, no need to activate Vyper
+    if (vyperTargets.length === 0) {
+      return { compilations: [] };
+    }
+
+    //having gotten the sources from the resolver, we invoke compileJson
+    //ourselves, rather than going through Compile.sources()
+    Compile.display(compilationTargets, options);
+
+    const { version, json: useJson } = await checkVyper();
+
+    if (useJson) {
+      return compileJson({
+        sources: allSources,
+        options: options.with({
+          compilationTargets
+        }),
+        version
+      });
+    } else {
+      return await compileNoJson({
+        paths: Object.keys(allSources),
+        options: options.with({
+          compilationTargets
+        }),
+        version
+      });
+    }
   },
 
   // contracts_directory: String. Directory where contract files can be found.
   // quiet: Boolean. Suppress output. Defaults to false.
   // strict: Boolean. Return compiler warnings as errors. Defaults to false.
   async all(options) {
     options = Config.default().merge(options);
-    const fileSearchPattern = path.join(
-      options.contracts_directory,
-      VYPER_PATTERN
+    const files = await findContracts(options.contracts_directory);
+
+    const vyperFilesStrict = files.filter(path =>
+      minimatch(path, VYPER_PATTERN_STRICT, { dot: true })
     );
-    debug("fileSearchPattern: %O", fileSearchPattern);
-    const files = await findContracts(fileSearchPattern);
-    debug("files: %O", files);
+    // no vyper targets found, no need to check Vyper
+    if (vyperFilesStrict.length === 0) {
+      return { compilations: [] };
+    }
 
-    return await Compile.sources({
-      sources: files,
+    return await Compile.sourcesWithDependencies({
```

Comment:
We change this to account for JSON imports, right?

Reply:
Not just JSON imports, but any imports that aren't in the contracts folder, same as with Solidity.  E.g. if you want to import Vyper from an NPM module -- with this PR, you can now do that!  But of course that requires getting the dependencies...